assessment-api/ddl.sql

```diff
@@ -0,0 +1,227 @@
+-- Prototype database design for Cabability assessment
+-- load tables
+
+create table cap_load_resources
+(resource_cluster varchar(100),
+ resource_family varchar(100));
+
+create table cap_load_service_capability
+(service_capability_hdr varchar(100),
+ service_capability     varchar(100));
+
+create table cap_load_jobs
+(role_cluster varchar(100),
+ role_num     varchar(10),
+ role_family  varchar(100),
+ role_name    varchar(100),
+ role_level   varchar(100));
+
+
+
+-- Create table dimensions
+
+
+create table dimensions 
+( dimension_name               varchar(30)   primary key,
+  dimension_descr              varchar(2000) not null,
+  min_weighting_pct            decimal,
+  max_weighting_pct            decimal,
+  allow_multiple_selection_ind boolean,
+  min_allowed_value            decimal,
+  max_allowed_value            decimal,
+  created_by                   varchar(2000) not null,                                   
+  created_at                   timestamp     not null,
+  updated_by                   varchar(2000) ,
+  updated_at                   timestamp);
+
+-- Create table cap_dimension_valid_values
+
+create table dimension_valid_values
+(dimension_name    varchar(30)   not null,
+ valid_value_code  varchar(30)   not null,
+ valid_value_name  varchar(30)   not null,
+ valid_value_descr varchar(2000) not null,
+ created_by        varchar(2000) not null,                                   
+ created_at        timestamp     not null,
+ updated_by        varchar(2000),
+ updated_at        timestamp,
+ primary key (dimension_name, valid_value_code));
+
+-- Create table submission_types
+
+create table submission_types 
+( submission_type_code  varchar(30)   primary key,
+  submission_type_name  varchar(100)  not null,
+  submission_type_descr varchar(2000) not null,
+  created_by            varchar(2000) not null,                                   
+  created_at            timestamp     not null,
+  updated_by            varchar(2000),
+  updated_at            timestamp);
+ 
+
+
+-- Create table assessment submission types
+
+create table assessment_submission_types 
+( assessment_submission_type_id       serial primary key,
+  assessment_tool_id    integer       not null,
+  submission_type_code  varchar(30)   not null,
+  created_by            varchar(2000) not null,                                   
+  created_at            timestamp     not null,
+  updated_by            varchar(2000),
+  updated_at            timestamp);
+
+
+-- Create table Assessment Tools - Originally know as Taxonomies
+
+create table assessment_tools
+( assessment_tool_id    integer       primary key,
```

Comment:
As discussed - assuming we don't want to use `assessment_tool_id` as the AgreementsService identifier - do we use `assessment_tool_name` or introduce a new field - some sort of fixed external identifier?

My vote would probably be for a new identifier, but happy to be downvoted

Reply:
Do you mean something like CA Number||Lot Number||assessment_name so RM6263 Lot 1 for a capability assessment would have the internal identifier of "RM6263Lot 1Capability"

Reply:
It could be anything really I guess - just as long as it is quite static (i.e. not a generated serial id or name that could be prone to change)

I'd probably keep it without spaces and not camel case (or substitute dashes or underscores - could just be `capability` even, or a number - not much good with names. Any preference @tommyb82 ?

Reply:
I'll create the column and then we can decide what is actually stored afterwards.